pkg/bitcoin/electrum/electrum_integration_test.go

```diff
@@ -295,6 +301,88 @@ func TestGetBlockHeader_Negative_Integration(t *testing.T) {
 	}
 }
 
+func TestGetTransactionMerkleProof_Integration(t *testing.T) {
+	transactionHash, err := bitcoin.NewHashFromString(
+		"72e7fd57c2adb1ed2305c4247486ff79aec363296f02ec65be141904f80d214e",
+		bitcoin.InternalByteOrder,
+	)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	blockHeight := uint(1569342)
+
+	expectedResult := &bitcoin.TransactionMerkleProof{
+		BlockHeight: 1569342,
+		MerkleNodes: []string{
+			"8b5bbb5bdf6727bf70fad4f46fe4eaab04c98119ffbd2d95c29adf32d26f8452",
+			"53637bacb07965e4a8220836861d1b16c6da29f10ea9ab53fc4eca73074f98b9",
+			"0267e738108d094ceb05217e2942e9c2a4c6389ac47f476f572c9a319ce4dfbc",
+			"34e00deec50c48d99678ca2b52b82d6d5432326159c69e7233d0dde0924874b4",
+			"7a53435e6c86a3620cdbae510901f17958f0540314214379197874ed8ed7a913",
+			"6315dbb7ce350ceaa16cd4c35c5a147005e8b38ca1e9531bd7320629e8d17f5b",
+			"40380cdadc0206646208871e952af9dcfdff2f104305ce463aed5eeaf7725d2f",
+			"5d74bae6a71fd1cff2416865460583319a40343650bd4bb89de0a6ae82097037",
+			"296ddccfc659e0009aad117c8ed15fb6ff81c2bade73fbc89666a22708d233f9",
+		},
+		Position: 176,
+	}
+
+	for testName, config := range testConfigs {
+		t.Run(testName, func(t *testing.T) {
+			electrum := newTestConnection(t, config.clientConfig)
+
+			result, err := electrum.GetTransactionMerkleProof(
+				transactionHash,
+				blockHeight,
+			)
+			if err != nil {
+				t.Fatal(err)
+			}
+
+			if diff := deep.Equal(result, expectedResult); diff != nil {
+				t.Errorf("compare failed: %v", diff)
+			}
+		})
+	}
+}
+
+func TestGetTransactionMerkleProof_Negative_Integration(t *testing.T) {
```

Comment:
There is something off about this test or mechanism. If you replace the URL in the config with something that will not work, for example, `s/blackie.c3-soft.com:57006/blackie.c3-softff.com:57006` all other negative integration tests fail with:
```
=== RUN   TestGetTransactionMerkleProof_Integration/fulcrum_ssl
    electrum_integration_test.go:439: failed to initialize electrum client: [retry timeout [1m0s] exceeded; most recent error: [dial tcp: lookup blackie.c3-softff.com: no such host]]
```

and this test hangs forever:
```
=== RUN   TestGetTransactionMerkleProof_Negative_Integration/fulcrum_ssl
panic: test timed out after 10m0s

goroutine 448 [running]:
testing.(*M).startAlarm.func1()
        /opt/homebrew/Cellar/go@1.18/1.18.5/libexec/src/testing/testing.go:2029 +0x8c
created by time.goFunc
        /opt/homebrew/Cellar/go@1.18/1.18.5/libexec/src/time/sleep.go:176 +0x3c

goroutine 1 [chan receive]:
testing.(*T).Run(0x14000582b60, {0x102aee146?, 0x1959a2c7808de?}, 0x102e1d870)
(...)
```

I am concerned we may have the client stuck with no error if the Electrum server stops responding.

Reply:
@pdyraga 
This problem was observed in other tests too and an issue was created to track it:
https://github.com/keep-network/keep-core/issues/3586